bin/blue_green_switch

```diff
@@ -18,29 +16,74 @@ class AWSClient
 end
 
 
-alb = AWSClient.elbv2.describe_load_balancers.load_balancers.find { |x| x.load_balancer_name == lb_name }
-listener = AWSClient.elbv2.describe_listeners(load_balancer_arn: alb.load_balancer_arn).listeners.find { |x| x.port == lb_port }
-rule = AWSClient.elbv2.describe_rules(listener_arn: listener.listener_arn).rules.find { |x| x.conditions.first.values.include? domain }
-tg_green = AWSClient.elbv2.describe_target_groups.target_groups.find { |x| x.target_group_name == "#{tg_prefix}-green" }
-tg_blue = AWSClient.elbv2.describe_target_groups.target_groups.find { |x| x.target_group_name == "#{tg_prefix}-blue" }
-
-if color == "green"
+if color_port == nil
+  fail "Cannot detect current color_port. Color_port not passed in cli?"
+elsif color == "green"
   puts "Switching to green"
-  tg_color = tg_green.target_group_arn
 elsif color == "blue"
   puts "Switching to blue"
-  tg_color = tg_blue.target_group_arn
 else
   fail "Cannot detect current color. Color not passed in cli?"
 end
 
-targets = AWSClient.elbv2.describe_target_health(target_group_arn: tg_color)
+tg = AWSClient.elbv2.describe_target_groups.target_groups.find { |x| x.target_group_name == "#{tg_name}" }
+targets = AWSClient.elbv2.describe_target_health(target_group_arn: tg.target_group_arn)
             .target_health_descriptions.map(&:target)
+targets_ids = targets.map(&:id)
+targets_port = targets.map(&:port).uniq[0]
```

Comment:
use `first` instead of [0] maybe?

Reply:
also maybe run riubocop against this script for styling? will solve most of this type of stuff

Reply:
Changed to `.first` and then removed it cause it's not needed :)
Checked with rubocop also and JFYI: rubocopo does not care about `[0]` vs `.first` even with default rules.